coinstacks/common/pulumi/src/rabbitmq.ts

```diff
@@ -0,0 +1,106 @@
+import * as pulumi from '@pulumi/pulumi'
+import * as k8s from '@pulumi/kubernetes'
+import { CustomResource } from '@pulumi/pulumi'
+import { rabbitmq, types } from './crds/types/rabbitmq'
+import { Config } from './index'
+
+export interface RabbitConfig {
+  cpuLimit: string
+  memoryLimit: string
+  replicaCount: number
+  storageClass: 'gp2' | 'hostpath' | 'standard'
+  storageSize: string
+}
+
+export async function deployRabbit(
+  app: string,
+  asset: string,
+  provider: k8s.Provider,
+  namespace: string,
+  config: Pick<Config, 'rabbit'>
+  ): Promise<CustomResource | undefined> {
+
```

Comment:
There is typically a check `if (config.rabbit === undefined) return` to not deploy the rabbit cluster if not defined depending on config requirement.